### PR TITLE
Use PYMAJOR/PYMINOR according to Python devguide

### PR DESCRIPTION
**Pull request by pmp-p**
also fixes hostpython paths so that pyodide can be built everywhere a clang based hostpython is available

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,21 +5,20 @@ include Makefile.envs
 .PHONY=check
 
 FILEPACKAGER=$$EM_DIR/tools/file_packager.py
-UGLIFYJS=npx uglifyjs
-PRETTIER=npx prettier
```
Comment (rth) [inline Makefile:9]: Thanks a lot @pmp-p ! Could you please merge upstream/main in as I think this was added just a few commits ago on main.

**Comment by pmp-p** [comment]
@rth indeed there were changes, looks ok now i think my edited makefile was from before renaming the main branch !

File: pyodide_build/buildpkg.py [hunk 2]
```diff
@@ -171,7 +171,8 @@ def compile(path: Path, srcpath: Path, pkg: Dict[str, Any], args):
 
     post = pkg.get("build", {}).get("post")
     if post is not None:
-        site_packages_dir = srcpath / "install" / "lib" / "python3.8" / "site-packages"
+        pyfolder = ''.join(["python", os.environ.get("PYMAJOR","3"), ".", os.environ.get("PYMINOR","8")])
+        site_packages_dir = os.path.sep.join(srcpath, "install", "lib", pyfolder, "site-packages" )
```
Comment (rth) [inline pyodide_build/buildpkg.py:175]: I think you would need
```suggestion
        site_packages_dir = os.path.join(srcpath, "install", "lib", pyfolder, "site-packages" )
```
or pass a list of paths

**Review by rth — APPROVED** [review]
Thanks a lot @pmp-p ! Could you please run `black` on `pyodide_build/buildpkg.py` and apply the fix below, to hopefully make CI pass? Otherwise LGTM.

File: pyodide_build/buildpkg.py [hunk 3]
```diff
@@ -171,7 +171,8 @@ def compile(path: Path, srcpath: Path, pkg: Dict[str, Any], args):
 
     post = pkg.get("build", {}).get("post")
     if post is not None:
-        site_packages_dir = srcpath / "install" / "lib" / "python3.8" / "site-packages"
+        pyfolder = ''.join(["python", os.environ.get("PYMAJOR","3"), ".", os.environ.get("PYMINOR","8")])
```
Comment (rth) [inline pyodide_build/buildpkg.py:174]: Maybe we could add

```suggestion
        # use Python, 3.8 by default
        pyfolder = ''.join(["python", os.environ.get("PYMAJOR","3"), ".", os.environ.get("PYMINOR","8")])
```
so we can find it by grep the day we change the default python version.

**Comment by pmp-p** [comment]
@rth please do

**Comment by hoodmane** [comment]
@rth What's the status on this? Can we merge it?

**Comment by rth** [comment]
I still need to fix CI. Something is not quite right. Looking into it.

**Pull request merged**